sdk/attestation/azure-security-attestation/src/main/java/com/azure/security/attestation/AttestationAsyncClient.java

```diff
@@ -8,43 +8,110 @@
+     * @throws RuntimeException all other wrapped checked exceptions if the request fails to be sent.
+     * @return any object.
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public Mono<Object> getOpenIdMetadata() {
+        // Forward the getOpenIdMetadata to the getOpenIdMetadataWithResponse API implementation.
+        return withContext(context -> this.getOpenIdMetadataWithResponse(context))
+            .map(Response::getValue);
+    }
+
+    /**
+     * Retrieves the list of {@link AttestationSigner} objects associated with this attestation instance.
+     * An {@link AttestationSigner} represents an X.509 certificate chain and KeyId which can be used
+     * to validate an attestation token returned by the service.
+     *
+     * @return Returns an array of {@link AttestationSigner} objects.
+     */
+
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public Mono<AttestationSigner[]> getAttestationSigners() {
+        return withContext(context -> this.getAttestationSignersWithResponse(context))
+            .map(Response::getValue);
+    }
+
+    /**
+     * Retrieves the list of {@link AttestationSigner} objects associated with this attestation instance.
+     *
+     * An {@link AttestationSigner} represents an X.509 certificate chain and KeyId which can be used
+     * to validate an attestation token returned by the service.
+     *
+     * @param context Context for the operation.
+     * @return Returns an array of {@link AttestationSigner} objects.
      */
-    AttestationAsyncClient(AttestationsImpl serviceClient) {
-        this.serviceClient = serviceClient;
+    public Mono<Response<AttestationSigner[]>> getAttestationSignersWithResponse(Context context) {
+        return this.signerImpl.getWithResponseAsync(context)
+            .map(response -> Utilities.generateResponseFromModelType(response, Utilities.attestationSignersFromJwks(response.getValue())));
     }
 
     /**
      * Processes an OpenEnclave report , producing an artifact. The type of artifact produced is dependent upon
      * attestation policy.
      *
      * @param request Attestation request for Intel SGX enclaves.
+     * @param context - Context for the operation
      * @throws IllegalArgumentException thrown if parameters fail the validation.
      * @throws CloudErrorException thrown if the request is rejected by server.
      * @throws RuntimeException all other wrapped checked exceptions if the request fails to be sent.
      * @return the result of an attestation operation.
      */
-    @ServiceMethod(returns = ReturnType.SINGLE)
-    public Mono<Response<AttestationResponse>> attestOpenEnclaveWithResponse(AttestOpenEnclaveRequest request) {
-        return this.serviceClient.attestOpenEnclaveWithResponseAsync(request.toGenerated())
+//    @ServiceMethod(returns = ReturnType.SINGLE)
+    public Mono<Response<AttestationResponse>> attestOpenEnclaveWithResponse(AttestOpenEnclaveRequest request, Context context) {
```

Comment:
>AttestOpenEnclaveRequest [](http://example.com/codeflow?start=77&length=24)

Any reason why you did not export this type directly in autorest,md - if you did that - both your implementation code and public code will depend on the public model so you would not have to call Todenerated method

Example: [https://github.com/Azure/azure-sdk-for-java/blob/main/sdk/containerregistry/azure-containers-containerregistry/swagger/autorest.md](https://github.com/Azure/azure-sdk-for-java/blob/main/sdk/containerregistry/azure-containers-containerregistry/swagger/autorest.md)

Specifically using
custom-types: and custom-types-subpackage: properties.

Reply:
That's a good point, and I guess I could have done that, but when I'm done with the SDK, none of the generated model types (except for 3 or 4 enumerations) are going to survive. 

By building clones of the generated types, it helps me track the conversion (it also means that when I get around to recreating the contents of a model type (like AttestationResult), I already have the skeleton of the type available.

For AttestOpenEnclaveRequest, that type is going to go away in the next pull request.

Reply:
As long as it won't survive and you are tracking it some way - I am good. This will have performance implications otherwise.

Reply:
Yes, I'm tracking it - I know exactly which types I expect to survive (there's about 5 or 6 of them), all the others will be thrown away (for instance, all the "*Result" types disappear quickly).